src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -195,7 +200,14 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
         final Set<String> rsIDs = getRSIDs(featureContext);
 
         if (VALIDATE_GVCF) {
-            genomeLocSortedSet.add(genomeLocSortedSet.getGenomeLocParser().createGenomeLoc(ref.getInterval().getContig(), ref.getInterval().getStart(), vc.getEnd()), true);
+            final SimpleInterval refInterval = ref.getInterval();
+            // Take advantage of adjacent blocks and just merge them so we dont have to keep so many objects in the set.
+            if (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 1)) {
```

Comment:
I think there's some pre-optimizing going on here. Please create `currentInterval` as a local final variable

Use it for checking overlap and for merging or replacing as needed.